frontend/src/components/checkbox/checkBox.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { ChangeEventHandler, ForwardedRef } from 'react';
+import { FieldError } from 'react-hook-form';
+import { Link } from 'react-router-dom';
+import { Checkbox, CheckboxText, CheckboxWrapper } from './checkbox-styles';
+import { Routes } from '../../enums/routes';
+import { Error } from '../input/input-styles';
+
+type CheckBoxProps = {
+  onChange?: ChangeEventHandler;
+  error?: FieldError;
+  name?: string;
+  type?: string;
+  label?: string;
+  link?: string;
+};
+
+const CheckBox = React.forwardRef(
+  (
+    { error, name, label, link, ...restProps }: CheckBoxProps,
+    ref: ForwardedRef<HTMLInputElement>
+  ) => (
+    <>
+      <CheckboxWrapper>
```

Comment:
Check how checkbox work semantically (in HTML). I'm pretty sure they don't have a "div" or a "paragraph"

Reply:
I am looking it up, how to make it reusable and make sense